### PR TITLE
feature CRA/2 :navbar menu plus logo

### PR DESCRIPTION
**Pull request by JRozbicka**

### CODE REVIEW

File: src/components/Navbar/MenuItems.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import * as FaIcons from "react-icons/fa";
+
+export const MenuItems = [
+  {
+    title: "Moje konto",
+    url: "#",
+    icon: <FaIcons.FaUserCog />,
```
Comment (JustD1d1t) [inline src/components/Navbar/MenuItems.js]: where can i find library for those icons?

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import Logo from "../logo/Logo";
+import { MenuItems } from "./MenuItems";
+import styles from "./Navbar.module.scss";
+
+// import * as FaIcons from "react-icons/fa";
```
Comment (JustD1d1t) [inline src/components/Navbar/Navbar.js]: Remove commented line

File: src/components/Navbar/Navbar.module.scss [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      transition:all 0.5s ease;
+    }
+    
+
+&__logo-container {
+   margin-left:0px;
+    background-size: cover;
+width: 200px;
+height: 200px;
+padding: 6px;
```
Comment (JustD1d1t) [inline src/components/Navbar/Navbar.module.scss]: Styling. Once time "tab", next time "tab + space", another nothing

File: src/components/Navbar/Navbar.module.scss [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+ @media screen and(max-width:960px){
+   margin-top: 20px;
+ }
+}
+
+    &__logo {
```
Comment (JustD1d1t) [inline src/components/Navbar/Navbar.module.scss]: same as above

File: src/components/Navbar/Navbar.module.scss [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  height: 150%;
+  object-fit: contain;
+
+  @media screen and(max-width:960px){
+    width: 100%;
+    height: 100%;
+    border-radius: 50%;
+   
+  }
+  }
```
Comment (JustD1d1t) [inline src/components/Navbar/Navbar.module.scss]: same as above

File: src/components/Navbar/Navbar.module.scss [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    justify-content: end;
+
+ @media screen and(max-width:960px){
+   text-align:center;
+          padding:32px;
+          width:100%;
+          display:table;
+       
+
+        }
```
Comment (JustD1d1t) [inline src/components/Navbar/Navbar.module.scss]: same as above

File: src/components/Navbar/Navbar.module.scss [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    grid-template-columns:repeat(
+        4,auto
+    );
+    grid-gap:15px;
+    list-style:none;
+    text-align:center;
+    justify-content: end;
+
+ @media screen and(max-width:960px){
+   text-align:center;
```
Comment (JustD1d1t) [inline src/components/Navbar/Navbar.module.scss]: same as above

File: src/components/Navbar/Navbar.module.scss [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+      
+
+ 
+ 
+
+
+
+
+
+
```
Comment (JustD1d1t) [inline src/components/Navbar/Navbar.module.scss]: remove empty lines

File: src/components/logo/Logo.jsx [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import logo from "./3.png";
+
+const Logo = (props) => {
+  const { classX } = props;
```
Comment (JustD1d1t) [inline src/components/logo/Logo.jsx]: change this wroking props name. I don't know what does it mean. Is it special class cause it's "X"?

**Review by JustD1d1t — COMMENTED** [review]
Resize this image. It is used only for logo so it should be much smaller. Use paint for example to resize this. I think max 200x200px should be fine

File: src/components/Navbar/MenuItems.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import * as FaIcons from "react-icons/fa";
```
Comment (ruljin) [inline src/components/Navbar/MenuItems.js]: The `package.json` file with `react-icons` as a dependency hasn't been added in any commits. Second, what is the point of importing the `whole` library into the file if you are using only those icons you need from it (`import { A, B, C } from 'react-icons/fa'`) and not the whole library (`... * as FaIcons from ...`).

File: src/components/Navbar/MenuItems.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
```
Comment (ruljin) [inline src/components/Navbar/MenuItems.js]: There is no need to import the whole react library into the file from the newest version of it.

File: src/components/logo/Logo.jsx (hunk 9 above)
Comment (ruljin) [inline src/components/logo/Logo.jsx]: Props validation check it up as missing. First of all why you don't destructure it when receiving it into the component?
`Logo = ({ classX }) => { ...`
Why just don't pass it for example as `classes`? I think it is more obvious and verbose that way.

File: src/components/logo/Logo.jsx [hunk 12]
```diff
@@ -0,0 +1,8 @@
+import logo from "./3.png";
```
Comment (ruljin) [inline src/components/logo/Logo.jsx]: The image should be moved to `/src/assets` and it should have an obvious name like for example `logo` or `logo__image`.

File: src/components/logo/Logo.jsx (hunk 9 above)
Comment (ruljin) [reply src/components/logo/Logo.jsx]: From what I'm aware you are using in the project `CSS modules` so why there is a `logo.css` file in addition that is empty? Remove it if it is empty.

File: src/components/Navbar/MenuItems.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+import * as FaIcons from "react-icons/fa";
+
+export const MenuItems = [
```
Comment (ruljin) [inline src/components/Navbar/MenuItems.js]: I have a few suggestions here. First of all, the file should be called `MenuItem` and have `.jsx` format. In shortcut, it should be a component that has the code for a single menu list item component.

Second, the `MenuItems` constant should be called `MENU_ITEMS` and should be moved to the `Navbar` component.

File: src/components/Navbar/MenuItems.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+  {
+    title: "Moje konto",
+    url: "#",
```
Comment (ruljin) [inline src/components/Navbar/MenuItems.js]: As far as I know, you already have routing for most pages and even if some is missing, you could ask your team leader to answer you under what URL would be a certain page available to already set it.

File: src/components/Navbar/Navbar.js [hunk 15]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
```
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: Same as before. There is no need to import the whole react library into the file from the newest version of it.

File: src/components/Navbar/Navbar.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+import Logo from "../logo/Logo";
+import { MenuItems } from "./MenuItems";
+import styles from "./Navbar.module.scss";
```
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: If I'm correct the `CSS module` name convention is `component_name_without_uppercase.module.scss`.

File: src/components/Navbar/Navbar.js (hunk 2 above)
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: Remove the comment.

File: src/components/Navbar/Navbar.js [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+
+function Navbar() {
+  return (
+    <nav className={styles.navbar}>
+      <div className={`${styles["navbar__logo-container"]}`}>
+        <Logo classX={`${styles.navbar__logo}`} />
+      </div>
+
+      <ul className={styles.navbar__items}>
+        {MenuItems.map((item, inx) => {
```
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: Avoid using shortcuts in variable names. It makes code harder to read. So no `inx` but `index`.

File: src/components/Navbar/Navbar.js [hunk 18]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import Logo from "../logo/Logo";
+import { MenuItems } from "./MenuItems";
+import styles from "./Navbar.module.scss";
+
+// import * as FaIcons from "react-icons/fa";
+
+function Navbar() {
```
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: As for me, it is better to use whenever it is possible a named export, not default.

File: src/components/Navbar/Navbar.js [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <nav className={styles.navbar}>
+      <div className={`${styles["navbar__logo-container"]}`}>
+        <Logo classX={`${styles.navbar__logo}`} />
+      </div>
+
+      <ul className={styles.navbar__items}>
```
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: The list should be granulated (moved) to the `Menu` component.

File: src/components/Navbar/Navbar.js [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+import Logo from "../logo/Logo";
+import { MenuItems } from "./MenuItems";
+import styles from "./Navbar.module.scss";
+
+// import * as FaIcons from "react-icons/fa";
+
+function Navbar() {
+  return (
+    <nav className={styles.navbar}>
+      <div className={`${styles["navbar__logo-container"]}`}>
```
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: Why using string literal here?

Comment (ruljin) [inline src/components/Navbar/Navbar.js]: Why using string literal here?

File: src/components/Navbar/Navbar.js (hunk 18 above)
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: As for me, it is better to use whenever it is possible a named export, not default.

File: src/components/Navbar/Navbar.js (hunk 20 above)
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: Why use a string literal here?

File: src/components/Navbar/Navbar.js [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+        <Logo classX={`${styles.navbar__logo}`} />
```
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: Why use a string literal here?

File: src/components/Navbar/Navbar.js [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+      </div>
+
+      <ul className={styles.navbar__items}>
+        {MenuItems.map((item, inx) => {
+          return (
+            <li keys={inx}>
```
Comment (ruljin) [inline src/components/Navbar/Navbar.js]: Same as before. Avoid using shortcuts in variable names - `index` not `inx`.

File: src/components/Navbar/Navbar.module.scss [hunk 23]
```diff
@@ -0,0 +1,109 @@
+.navbar {
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: You have `prettier` configuration in the project. Why not format code using it?

File: src/components/Navbar/Navbar.module.scss [hunk 24]
```diff
@@ -0,0 +1,109 @@
...
+    font-size: 34px;
+    padding: 0 30px;
+    @media screen and(max-width:960px) {
+       display:flex;
+      flex-direction:column;
+      width:100%;
+      height:500px;
+      transition:all 0.5s ease;
+    }
+
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: Max and minimal one empty line of space between rules.

File: src/components/Navbar/Navbar.module.scss [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
+  border-radius: 50%;
+  width: 150%;
+  height: 150%;
+  object-fit: contain;
+
+  @media screen and(max-width:960px){
+    width: 100%;
+    height: 100%;
+    border-radius: 50%;
+
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: Max and minimal one empty line of space between rules.

File: src/components/Navbar/Navbar.module.scss [hunk 26]
```diff
@@ -0,0 +1,109 @@
...
+    list-style:none;
+    text-align:center;
+    justify-content: end;
+
+ @media screen and(max-width:960px){
+   text-align:center;
+          padding:32px;
+          width:100%;
+          display:table;
+
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: Max and minimal one empty line of space between rules.

File: src/components/Navbar/Navbar.module.scss [hunk 27]
```diff
@@ -0,0 +1,109 @@
...
+    text-align:center;
+    justify-content: end;
+
+ @media screen and(max-width:960px){
+   text-align:center;
+          padding:32px;
+          width:100%;
+          display:table;
+       
+
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: Max and minimal one empty line of space between rules.

File: src/components/Navbar/Navbar.module.scss [hunk 28]
```diff
@@ -0,0 +1,109 @@
...
+ &__span {
+  margin-left: 16px;
+  font-size: 30px;
+@media screen and(max-width:960px){
+     font-size: 20px;
+   }
+}
+
+}
+
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: Max and minimal one empty line of space between rules.

File: src/components/logo/Logo.jsx [hunk 29]
```diff
@@ -0,0 +1,8 @@
...
+
+const Logo = (props) => {
```
Comment (ruljin) [inline src/components/logo/Logo.jsx]: As for me, it is better to use whenever it is possible a named export, not default.

**Review by ruljin — CHANGES_REQUESTED** [review]
There is still some room for improvement.

File: src/components/Navbar/Navbar.module.scss [hunk 30]
```diff
@@ -0,0 +1,109 @@
+.navbar {
+    background: #c9c9f1;
+    height: 250px;
+   width: 100%;
+    position: fixed;
```
Comment (JustD1d1t) [inline src/components/Navbar/Navbar.module.scss]: i think it should be "sticky"

File: src/assets/logo/Logo.jsx [hunk 31]
```diff
@@ -0,0 +1,5 @@
+import logo from "./logo__image.png";
+
+export const Logo = ({ classes }) => {
```
Comment (ruljin) [inline src/assets/logo/Logo.jsx]: `Logo` **image file** should be in **assets directory** and it should be **imported** to the `Logo` **component file** which should be in **components directory**.

File: src/components/Navbar/MenuItem.jsx [hunk 32]
```diff
@@ -0,0 +1,3 @@
+export const MenuItem = ({ children }) => {
+  return { children };
```
Comment (ruljin) [inline src/components/Navbar/MenuItem.jsx]: What is the point of creating a **component** that the only thing it does is, display the same element it receives? It is the same as displaying it on the parent level but longer.

**Review by ruljin — CHANGES_REQUESTED** [review]
Changes requested.

File: src/components/Navbar/Menu.jsx [hunk 33]
```diff
@@ -0,0 +1,33 @@
+import { Link } from "react-router-dom";
+import propTypes from "prop-types";
+import styles from "./Navbar.module.scss";
+import { Logo } from "../Logo/Logo";
+
+export const Menu = ({ list }) => {
+  return (
+    <div className={styles[("navbar__logo-container", "menu")]}>
```
Comment (ruljin) [inline src/components/Navbar/Menu.jsx]: You can install and use `clsx`

File: src/components/Navbar/Menu.jsx [hunk 34]
```diff
@@ -0,0 +1,33 @@
...
+      {list.map((element) => {
```
Comment (ruljin) [inline src/components/Navbar/Menu.jsx]: Not `element` but `item` or `listItem`

File: src/components/Navbar/Menu.jsx [hunk 35]
```diff
@@ -0,0 +1,33 @@
+import { Link } from "react-router-dom";
+import propTypes from "prop-types";
```
Comment (ruljin) [inline src/components/Navbar/Menu.jsx]: Be consistent `PropTypes`

File: src/components/Navbar/Menu.jsx [hunk 36]
```diff
@@ -0,0 +1,33 @@
...
+import styles from "./Navbar.module.scss";
```
Comment (ruljin) [inline src/components/Navbar/Menu.jsx]: Module names from a lowercase

File: src/components/Navbar/Navbar.jsx [hunk 37]
```diff
@@ -0,0 +1,37 @@
+// import { useEffect, useState } from "react";
```
Comment (ruljin) [inline src/components/Navbar/Navbar.jsx]: Remove it

File: src/components/Navbar/Navbar.jsx [hunk 38]
```diff
@@ -0,0 +1,37 @@
...
+import styles from "./Navbar.module.scss";
```
Comment (ruljin) [inline src/components/Navbar/Navbar.jsx]: Module names from a lowercase

File: src/components/Navbar/Navbar.jsx [hunk 39]
```diff
@@ -0,0 +1,37 @@
...
+import { routes } from "../../routes/index";
+import { Menu } from "./Menu";
+
+const Hamburger = () => (
+  <div className={styles.hamburger} id={styles.hamburger}>
```
Comment (ruljin) [inline src/components/Navbar/Navbar.jsx]: Is there any reason for this `id` here?

File: src/components/Navbar/Navbar.jsx [hunk 40]
```diff
@@ -0,0 +1,37 @@
...
+    <span className={styles.line} />
+    <span className={styles.line} />
+    <span className={styles.line} />
+  </div>
+);
+
+export const Navbar = () => {
+  // commented for now cause it might be useful
```
Comment (ruljin) [inline src/components/Navbar/Navbar.jsx]: Remove the whole comment `!!!`

File: src/components/Navbar/Navbar.module.scss [hunk 41]
```diff
@@ -0,0 +1,157 @@
+// .navbar {
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: REMOVE THE WHOLE COMMENT `!!!`

File: src/components/Navbar/Navbar.module.scss [hunk 42]
```diff
@@ -0,0 +1,157 @@
...
+//     margin-left: 16px;
+//     font-size: 30px;
+
+//     @media screen and(max-width:960px) {
+//       font-size: 20px;
+//     }
+//   }
+// }
+
+//Hamburger styles
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: ☝️

File: src/components/Navbar/Navbar.module.scss [hunk 43]
```diff
@@ -0,0 +1,157 @@
...
+
+.hamburger:hover {
+  cursor: pointer;
+}
+
+#hamburger {
+  -webkit-transition: all 0.3s ease-in-out;
+  -o-transition: all 0.3s ease-in-out;
+  transition: all 0.3s ease-in-out;
+  // display: none;
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: ☝️

File: src/components/logo/Logo.jsx [hunk 44]
```diff
@@ -0,0 +1,8 @@
+import propTypes from "prop-types";
```
Comment (ruljin) [inline src/components/logo/Logo.jsx]: Be consistent - `PropTypes`

File: src/routes/index.js [hunk 45]
```diff
@@ -1,21 +1,54 @@
+import { FaRegIdCard, FaGratipay, FaUserCog, FaUserPlus } from "react-icons/fa";
 import { Auth } from "../modules/auth";
 import { Main } from "../modules/main";
 import { Account } from "../modules/account";
-import { announcementsRoutes } from "../modules/announcements";
+import { Favorites } from "../modules/announcements/pages/Favorites";
+import { AnnouncementList } from "../modules/announcements/pages/Announcements";
+import { AnnouncementNew } from "../modules/announcements/pages/New";
+// import { Announcement } from "../modules/announcements/pages/Announcement";
```
Comment (ruljin) [inline src/routes/index.js]: Remove it

File: src/routes/index.js [hunk 46]
```diff
@@ -1,21 +1,54 @@
...
+    icon: <FaRegIdCard />,
+  },
+  {
+    path: "/announcements/favorites",
+    component: <Favorites />,
+    title: "Ulubione",
+    icon: <FaGratipay />,
+    visibleWhenLogged: true,
+  },
+  // {
```
Comment (ruljin) [inline src/routes/index.js]: Remove it

**Review by ruljin — CHANGES_REQUESTED** [review]
Changes requested

File: src/components/Navbar/Navbar.module.scss [hunk 47]
```diff
@@ -0,0 +1,82 @@
+.navbar {
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: Git detect this file as `Navbar.module.scss` - wasn't it deleted?

File: src/components/logo/Logo.jsx [hunk 48]
```diff
@@ -0,0 +1,8 @@
+import PropTypes from "prop-types";
+import logo from "./logo.png";
```
Comment (ruljin) [inline src/components/logo/Logo.jsx]: Move it to assets and import it from it

**Review by ruljin — CHANGES_REQUESTED** [review]
Changed requested

File: src/components/Navbar/Menu.jsx [hunk 49]
```diff
@@ -0,0 +1,34 @@
+import { Link } from "react-router-dom";
+import PropTypes from "prop-types";
+import clsx from "clsx";
+import styles from "./navbar.module.scss";
+import { Logo } from "../logo/Logo";
+
+export const Menu = ({ list }) => {
+  return (
+    <div className={clsx([styles.navbar__logo_container, styles.menu])}>
```
Comment (ruljin) [inline src/components/Navbar/Menu.jsx]: There should be something like two or three containers on the menu. The `left` side and the `right` side. On the left side, there should be a logo and the navigation of the basic route. On the right side, there should be login and register URLs.

File: src/routes/index.js [hunk 50]
```diff
@@ -1,21 +1,44 @@
 import { Auth } from "../modules/auth";
 import { Main } from "../modules/main";
 import { Account } from "../modules/account";
-import { announcementsRoutes } from "../modules/announcements";
+import { Favorites } from "../modules/announcements/pages/Favorites";
+import { AnnouncementList } from "../modules/announcements/pages/Announcements";
+import { AnnouncementNew } from "../modules/announcements/pages/New";
 
 export const routes = [
```
Comment (ruljin) [inline src/routes/index.js]: The URLs titles are other on the design than in the routes.

File: src/components/Navbar/Navbar.module.scss [hunk 51]
```diff
@@ -0,0 +1,152 @@
...
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: Navbar styles are still detected as `navbar.module.scss`. It shouldn't exist anymore and the `navbar.module.scss` file should be detected. Besides, the styles for the desktop `Navbar` are not proper. The height is different, routes placement and padding between elements and from the page sides.

File: src/components/Navbar/Navbar.module.scss [hunk 52]
```diff
@@ -0,0 +1,152 @@
...
+  -webkit-transition: all 0.3s ease-in-out;
+  -o-transition: all 0.3s ease-in-out;
+  transition: all 0.3s ease-in-out;
+}
+
+.hamburger:hover {
+  cursor: pointer;
+}
+
+#hamburger {
```
Comment (ruljin) [inline src/components/Navbar/Navbar.module.scss]: I think that even for the mobile the whole `Navbar` shouldn't disappear. The only change should generate a `Navbar` with the logo on the left and with the hamburger icon (button) on the right, which opens and close the mobile menu.

**Review by ruljin — CHANGES_REQUESTED** [review]
Changes requested

File: src/components/Navbar/Navbar.jsx [hunk 53]
```diff
@@ -1,25 +1,24 @@
...
-      setChange(!change);
-      hamburgerClasses = styles.hamburger;
-    }
+    handleHamburgerClick(!isOpen);
   };
+
   return (
     <button
-      className={hamburgerClasses}
+      className={isOpen ? hamburgerClasses : styles.hamburger}
```
Comment (ruljin) [inline src/components/Navbar/Navbar.jsx]: Why not pass here only `hamburgerClasses`? You have `clsx` that manipulate styles depending on props value.

**Review by ruljin — CHANGES_REQUESTED** [review]
Changes requested

**Pull request merged**